DFE-Digital/npq-registration
# Reject funded NPQs

### Context

[Ticket CN-22](https://dfedigital.atlassian.net/browse/CN-22)

We want users to not be eligible for scholarship funding for the same NPQ, if they have already received one in the past.

### Changes proposed in this pull request

- Added the NpqFunding class to handle the api call
- Added a custom parser for the NpqFunding to parse the response from the ECF API
- Update the ineligibility copy
- Add an ineligibility page for the EHCO NPQ (a.k.a ASO)
- Update the funding eligibility service to require a TRN
- Added course mappings for the ECF course identifiers

### Guidance to review
The previously funded eligibility check is institution agnostic - will run for Schools, Early Years etc.

Also, the npq funding endpoint in the ECF API requires a TRN and a course identifier, but the NPQ service only knows the `ecf_id`. It make sense to update the ECF API at some point to require the `ecf_id` instead, but for now I have added a mapping in the `Course` model to resolve the `identifier` based on the `ecf_id`.

#### How to test
Run NPQ and ECF services on your local and go through an eligible journey. Choose users and NPQs that were funded in the past and then check the copy in the ineligibility page is correct.
